Controller/Admin/Transaction/TransactionTabPostProcessing.php

```diff
@@ -388,47 +392,55 @@ private function _getTranslatedPostProcessingActions($aSupportedOperations)
      */
     private function _handleRequestAction($sActionTitle, $fAmount)
     {
-        $oTransactionHandler = $this->_getTransactionHandler();
+        $oTransactionHandler = $this->_getTransactionHandler(true);
         $aResult = $oTransactionHandler->processAction($this->oTransaction, $sActionTitle, $fAmount);
 
         return $aResult["status"] === Transaction::STATE_SUCCESS ?
             Helper::translate('wd_text_generic_success') : $aResult['message'];
     }
 
     /**
-     * Returns an instance of TransactionHandler (singleton)
+     * Returns an instance of TransactionHandler
+     *
+     * @param bool $bPostProcessing
      *
      * @return TransactionHandler
      *
      * @since 1.0.1
      */
-    private function _getTransactionHandler()
+    private function _getTransactionHandler($bPostProcessing)
```

Comment:
Using a parameter (true/false) is usually avoided at all costs. You did not document why the flag is used. Why do you need this?

Reply:
But the only difference is you have another transaction type. If you make a method in the SofortPaymentMethod where you can get a transaction type for refund all your problems are solved the same is for SEPA DD

Reply:
I think you build a lot of overhead where its not needed.

Reply:
changed it as discussed

Reply:
changed it as discussed